docs/how-to/sampling-call-stack.rst

```diff
@@ -0,0 +1,380 @@
+showing which environment variables were set or modified:
+
+The snippet below shows the environment updates when ``omnitrace-sample`` is invoked with no arguments
+
+.. code-block:: shell
+
+   $ omnitrace-sample -- ./parallel-overhead-locks 30 4 100
+
+   HSA_TOOLS_LIB=/opt/omnitrace/lib/libomnitrace-dl.so.1.7.1
+   HSA_TOOLS_REPORT_LOAD_FAILURE=1
+   LD_PRELOAD=/opt/omnitrace/lib/libomnitrace-dl.so.1.7.1
+   OMNITRACE_USE_PROCESS_SAMPLING=false
+   OMNITRACE_USE_SAMPLING=true
+   OMP_TOOL_LIBRARIES=/opt/omnitrace/lib/libomnitrace-dl.so.1.7.1
+   ROCP_TOOL_LIB=/opt/omnitrace/lib/libomnitrace.so.1.7.1
+
+The snippet below shows the environment updates when ``omnitrace-sample`` enables 
+profiling, tracing, host process-sampling, device process-sampling, and all the available backends:
+
+.. code-block:: shell
+
+   $ omnitrace-sample -PTDH -I all -- ./parallel-overhead-locks 30 4 100
+
+   HSA_TOOLS_LIB=/opt/omnitrace/lib/libomnitrace-dl.so.1.7.1
+   HSA_TOOLS_REPORT_LOAD_FAILURE=1
+   KOKKOS_PROFILE_LIBRARY=/opt/omnitrace/lib/libomnitrace.so.1.7.1
+   LD_PRELOAD=/opt/omnitrace/lib/libomnitrace-dl.so.1.7.1
+   OMNITRACE_CPU_FREQ_ENABLED=true
+   OMNITRACE_TRACE_THREAD_LOCKS=true
+   OMNITRACE_TRACE_THREAD_RW_LOCKS=true
+   OMNITRACE_TRACE_THREAD_SPIN_LOCKS=true
+   OMNITRACE_USE_KOKKOSP=true
+   OMNITRACE_USE_MPIP=true
+   OMNITRACE_USE_OMPT=true
+   OMNITRACE_TRACE=true
+   OMNITRACE_USE_PROCESS_SAMPLING=true
+   OMNITRACE_USE_RCCLP=true
+   OMNITRACE_USE_ROCM_SMI=true
+   OMNITRACE_USE_ROCPROFILER=true
+   OMNITRACE_USE_ROCTRACER=true
+   OMNITRACE_USE_ROCTX=true
+   OMNITRACE_USE_SAMPLING=true
+   OMNITRACE_PROFILE=true
+   OMP_TOOL_LIBRARIES=/opt/omnitrace/lib/libomnitrace-dl.so.1.7.1
+   ROCP_TOOL_LIB=/opt/omnitrace/lib/libomnitrace.so.1.7.1
+   ...
+
+The snippet below shows the environment updates when ``omnitrace-sample`` enables 
+profiling, tracing, host process-sampling, and device process-sampling,
+sets the output path to ``omnitrace-output`` and the output prefix to ``%tag%``, and disables 
+all the available backends:
+
+.. code-block:: shell
+
+   $ omnitrace-sample -PTDH -E all -o omnitrace-output %tag% -- ./parallel-overhead-locks 30 4 100
+
+   LD_PRELOAD=/opt/omnitrace/lib/libomnitrace-dl.so.1.7.1
+   OMNITRACE_CPU_FREQ_ENABLED=true
+   OMNITRACE_OUTPUT_PATH=omnitrace-output
+   OMNITRACE_OUTPUT_PREFIX=%tag%
+   OMNITRACE_TRACE_THREAD_LOCKS=false
+   OMNITRACE_TRACE_THREAD_RW_LOCKS=false
+   OMNITRACE_TRACE_THREAD_SPIN_LOCKS=false
+   OMNITRACE_USE_KOKKOSP=false
+   OMNITRACE_USE_MPIP=false
+   OMNITRACE_USE_OMPT=false
+   OMNITRACE_TRACE=true
+   OMNITRACE_USE_PROCESS_SAMPLING=true
+   OMNITRACE_USE_RCCLP=false
+   OMNITRACE_USE_ROCM_SMI=false
+   OMNITRACE_USE_ROCPROFILER=false
+   OMNITRACE_USE_ROCTRACER=false
+   OMNITRACE_USE_ROCTX=false
+   OMNITRACE_USE_SAMPLING=true
+   OMNITRACE_PROFILE=true
+   ...
+
+An ``omnitrace-sample`` example
+========================================
+
```

Comment:
Below is the full output from the previously described example. 

You can probably word this better than me. :) I'm just giving this example a little bit of an intro. It is the same command-line as the example above;  ie. ` $ omnitrace-sample -PTDH -E all -o omnitrace-output %tag% -- ./parallel-overhead-locks 30 4 100`. This is just the full output.

Reply:
Sure. I'll work on this tomorrow and will get your input on the results. Thanks for all the feedback!